library/public/engine.h

```diff
@@ -21,38 +21,31 @@
 #ifndef f3d_engine_h
 #define f3d_engine_h
 
+#include "exception.h"
 #include "export.h"
```

Comment:
I'm honestly not sure about this change. There are some drawbacks doing this, like increasing the compilation time for the user. I still think forward decl is better.

Reply:
If we want to simplify usability, we can provide a f3d.h file which include all other files. This way, users who want to include only required headers can still do it.

Reply:
With less than 10 headers, I think there is no increase. Did you read the link provided by @Connor-Bowley ?

Reply:
Fair enough. I'm not used to that but I guess it's an old habit